### PR TITLE
Add export network trainable variables (PRETRAIN_VARS) to experiment dir

### LINKED ISSUE #1178: Need a list of PRETRAIN_VARS for fine tuning
Blueoil can use weight from a pre-train model but it is complicated to obtain pre-train variables list (`PRETRAIN_VARS`), e.g., which how to list up variables and which variable can be used.

### PR DESCRIPTION
**Pull request by oatawa1**
## What this patch does to fix the issue.
Add export network trainable variables (`PRETRAIN_VARS`) to the experiment directory.

Example:
Network: LmnetV1Quantize (https://docs.blue-oil.org/tutorial/image_cls.html)

`path/to/experiment/pretrain_vars.txt`
```
[
    'conv1/conv2d/kernel:0',
    'conv1/BatchNorm/beta:0',
    'conv1/BatchNorm/gamma:0',
    'conv2/conv2d/kernel:0',
    'conv2/BatchNorm/beta:0',
    'conv2/BatchNorm/gamma:0',
    'conv3/conv2d/kernel:0',
    'conv3/BatchNorm/beta:0',
    'conv3/BatchNorm/gamma:0',
    'conv4/conv2d/kernel:0',
    'conv4/BatchNorm/beta:0',
    'conv4/BatchNorm/gamma:0',
    'conv5/conv2d/kernel:0',
    'conv5/BatchNorm/beta:0',
    'conv5/BatchNorm/gamma:0',
    'conv6/conv2d/kernel:0',
    'conv6/BatchNorm/beta:0',
    'conv6/BatchNorm/gamma:0',
    'conv7/kernel:0',
    'conv7/bias:0',
]

```

## Link to any relevant issues or pull requests.
Closes #1178

### CODE REVIEW

File: blueoil/cmd/train.py [hunk 1]
```diff
@@ -122,6 +122,12 @@ def start_training(config, profile_step):
 
         saver = tf.compat.v1.train.Saver(max_to_keep=config.KEEP_CHECKPOINT_MAX)
 
+        with open(os.path.join(environment.EXPERIMENT_DIR, "pretrain_vars.txt"), 'w') as pretrain_vars_file:
+            train_vars = tf.compat.v1.trainable_variables()
+            pretrain_vars_file.writelines("[\n")
+            pretrain_vars_file.writelines("\t'%s',\n" % var.name for var in train_vars)
```
Comment (iizukak) [inline blueoil/cmd/train.py:128]: We will copy and paste this variables. So it's good to use space instead of tab.

**Review by iizukak — CHANGES_REQUESTED** [review]

File: blueoil/cmd/train.py (hunk 1 above)
Comment (oatawa1) [reply blueoil/cmd/train.py]: Thank you. I fixed it.

**Review by iizukak — APPROVED** [review]
OA

File: blueoil/cmd/train.py [hunk 2]
```diff
@@ -122,6 +122,12 @@ def start_training(config, profile_step):
 
         saver = tf.compat.v1.train.Saver(max_to_keep=config.KEEP_CHECKPOINT_MAX)
 
+        with open(os.path.join(environment.EXPERIMENT_DIR, "pretrain_vars.txt"), 'w') as pretrain_vars_file:
+            train_vars = tf.compat.v1.trainable_variables()
+            pretrain_vars_file.writelines("[\n")
+            pretrain_vars_file.writelines("    '%s',\n" % var.name for var in train_vars)
+            pretrain_vars_file.writelines("]\n")
```
Comment (tfujiwar) [inline blueoil/cmd/train.py:129]: It would be better to use `json.dumps` if you want to format the file as JSON.

Comment (tfujiwar) [reply blueoil/cmd/train.py:129]: ah, it's a part of python (config.py). Please ignore this comment.

**Review by tfujiwar — APPROVED** [review]
RA

**Pull request merged**